ui/src/app/groupings/[groupingPath]/_components/export-dropdown.tsx

```diff
@@ -1,16 +1,211 @@
 'use client';
 
-import { Button } from '@/components/ui/button';
-import { FileInput, ChevronDown } from 'lucide-react';
+import { useReducer, useRef } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+    faFileExport,
+    faCaretDown,
+    faUsers,
+    faUserPlus,
+    faUserMinus,
+    faIdCard
+} from '@fortawesome/free-solid-svg-icons';
+import { z } from 'zod';
+import { Tooltip, TooltipContent, TooltipProvider, TooltipTrigger } from '@/components/ui/tooltip';
+
+const memberSchema = z.object({
+    firstName: z.string(),
+    lastName: z.string(),
+    uid: z.string(),
+    uhUuid: z.string()
+});
+
+const groupingMemberSchema = z.object({
+    allMembers: z.object({
+        members: z.array(memberSchema)
+    }),
+    groupingInclude: z.object({
+        members: z.array(memberSchema)
+    }),
+    groupingExclude: z.object({
+        members: z.array(memberSchema)
+    }),
+    groupingBasis: z.object({
+        members: z.array(memberSchema)
+    })
+});
+
+type Member = z.infer<typeof memberSchema>;
+type GroupingMembers = z.infer<typeof groupingMemberSchema>;
+
+interface ExportDropdownProps {
+    groupingMembers: GroupingMembers;
+    groupPath: string;
+}
+
+const initialState = {
+    isOpen: false
+};
+
+type State = typeof initialState;
+type Action = { type: 'TOGGLE' } | { type: 'CLOSE' };
+
+const reducer = (state: State, action: Action): State => {
+    switch (action.type) {
+        case 'TOGGLE':
+            return { ...state, isOpen: !state.isOpen };
+        case 'CLOSE':
+            return { ...state, isOpen: false };
+    }
+};
```

Comment:
A reducer would not be necessary here, a reducer should be used because a single useState can handle all of the state management. You should use a reducer when you have many useStates and useEffects being used together, just like how you refactored the Description form component.